CONTRIBUTING.rst

```diff
@@ -63,6 +63,32 @@ a PR until it conforms to the requirements described here (e.g.
 coding style, release notes, etc.) and it is confirmed that the code
 has sufficient unit tests and does not break any existing unit tests.
 
+Commits
+-------
+
+While not a strict requirement, it is beneficial for commits to record a set of
+separate, logical changes.  Ideally, each commit should compile and have
+correct code style (this is partly enforced by our workflow tools, and such
+enforcement may be expanded in the future).  "Fixup" commits only serve to
+clutter the history and make it harder to understand what changed.  Judicious
+use of amend and rebase is encouraged.
+
+Additionally, commits should follow the accepted conventions for git commit
+messages.  In short:
+
+- Use proper spelling and grammar; avoid "twitter speak".
+- Use complete sentences (including the first line) and appropriate
+  capitalization.  Use imperative mood.
+- Try to limit the subject line to 50 characters.  *Don't* end with a period.
```

Comment:
Along the lines of the above comment, 'Try' might be too soft. Ghostflow will complain if the line is too long.

Reply:
I wasn't aware of that; that's a good argument for possibly stronger language. I'm hesitant to make this an absolute requirement, though, and I'm not otherwise coming up with an alternate phrasing that doesn't feel more clumsy. Suggestions welcomed? (BTW, does Ghostflow *not* complain about long body lines? Can we add that? ...probably should teach it an exception for lines with "words" longer than 60 characters, though.)

Reply:
I think there is always a way to make the subject line meet a hard maximum (either 50 or 60 characters). You have a lot of space following to explain what you meant ;-) Is there any line size restriction for the following text? Does it get automatically wrapped?

Reply:
> Is there any line size restriction for the following text?

Uh...

```
- Limit body lines (when necessary; occasionally just a subject is enough) to
  72 characters (unless this is impossible e.g. due to a long URL).
```

...yes? :wink:

> Does it get automatically wrapped?

That depends on your editor and if it's configured for git commit messages. Vim can be so configured and may work OOTB if it is your default editor. (Note that I'm unsure if "vim-minimal" is inadequate.)

Reply:
> I think there is always a way to make the subject line meet a hard maximum (either 50 or 60 characters).

Sure, if "make it unhelpful" is one of those ways¹. While I can't think of any instances offhand where I have *definitely* not found a 50-characters-or-less subject acceptable, I know I bump up against the limit some times, and I'm not 100% confident I've never opted to break it.

(¹ Hint: it isn't. Although "Fix typos" is probably acceptable, something like "Fix the bug" is not. I already think this section is longer than I'd prefer, and I don't want to copy all of what's in the cited articles, the subject should give some indication of what the commit does. Terse and descriptive can, naturally, be at odds.)